python/ray/data/_internal/logical/operators/read_operator.py

```diff
@@ -43,11 +45,41 @@ def get_detected_parallelism(self) -> int:
         """
         return self._detected_parallelism
 
-    def schema(self):
-        return self._datasource.schema()
+    def aggregate_output_metadata(self) -> BlockMetadata:
+        # Legacy datasources might not implement `get_read_tasks`.
```

Comment:
i think it would be good to add a comment clarifying that this does not trigger execution of the dataset.